symtabAPI/src/Object-elf.h

```diff
@@ -483,6 +529,12 @@ class ObjectELF final : public Object
   void parse_dynamicSymbols( Elf_X_Shdr *& dyn_scnp, Elf_X_Data &symdata,
                              Elf_X_Data &strdata, bool shared_library);
 
+  bool parse_riscv_attrs(std::string &, std::string &, std::function<int(std::string &, int)>);
```

Comment:
In/out parameters (especially stringy data) hurt my soul. Could this not just return the parsed attribute?

Reply:
The parsed attribute is handled in the lambda function (third argument). I discussed with Jim on Tuesday, and he suggested this function to be architecture agnostic. So I think we need at least three parameters for this function: the attribute string itself, the name of the attribute (which corresponds to the attribute we want to parse), and a function that parses the attribute data,  which is architecture dependent, and assign the parsed to the corresponding structure in ObjectELF). Since the parsed data is architecture dependent, I don't think returning parsed attribute is a good idea.

(By the way, since this function is architecture agnostic, I forgot to rename it to `parse_riscv_attrs` instead of `parse_attrs`).